package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -119,15 +149,12 @@ def validate_atom_types(self, atom_types):
                 except KeyError:
                     warnings.warn("Failed to guess the mass for the following atom types: {}".format(atom_type))
 
+
     def get_atom_mass(self, element):
         """Return the atomic mass in u for *element*.
-
         Masses are looked up in :data:`MDAnalysis.topology.tables.masses`.
+        .. Warning:: Unknown masses are set to nan
```

Comment:
We cannot do this change yet (the behaviour, not just the docstring). Indeed, this will change the behaviour of the library in a way that is not backwards compatible.

Here is what I suggest we do:
* restore the previous behaviour and use 0 when the mass cannot be guessed, this will be the behaviour for this PR (keep the ITP parser using nan, it is a different matter)
* in a later PR, we will add a flag to use `missing_value_label` for values that cannot be guessed. The flag will be off by default so the default behaviour remains unchanged
* add a deprecation warning about the flag to warn that version 3.0 will change the default value
* in version 3.0.0 we do the final switch

This is cumbersome but we have some responsibility toward the users to at least try not to break their stuff. When we do, we need to warn them beforehand and give them a way to smoothly upgrade. People rely on the library for their research.